lib/OpenQA/WebAPI/Controller/API/V1/Job.pm

```diff
@@ -219,60 +220,174 @@ sub overview {
+    my $job_id = $job->id;
+    my $json = $self->{_json};
+    (defined $job_suffix ? $json->{ids}->{$job_suffix} : $json->{id}) = $job_id;
+    push @{$self->{_event_data}}, {id => $job_id, %params};
+
+    # enqueue gru jobs and calculate blocked by
+    push @{$downloads->{$_}}, [$job_id] for keys %$downloads;
+    $self->gru->enqueue_download_jobs($downloads);
+    return $job_id;
+}
+
+sub _turn_suffix_into_job_id ($ids, $dep, $key) {
+    my $job_suffix = $dep->{$key};
+    die "Specified dependency '$job_suffix' does not relate to a present job-suffix.\n"
+      unless my $job_id = $ids->{$job_suffix};
+    $dep->{$key} = $job_id;
+}
+
+=over 4
+
+=item _create_dependencies()
+
+Creates dependencies from $self->{_dependencies} within the database.
+
+=back
+
+=cut
+
+sub _create_dependencies ($self) {
+    my $deps = $self->{_dependencies};
+    my $ids = $self->{_json}->{ids};
+    return undef unless @$deps;
+    my $job_dependencies = $self->schema->resultset('JobDependencies');
+    for my $dep (@$deps) {
+        _turn_suffix_into_job_id($ids, $dep, $_) for qw(child_job_id parent_job_id);
+        $job_dependencies->create($dep);
+    }
+    my $jobs = $self->schema->resultset('Jobs');
+    $jobs->find($_)->calculate_blocked_by for values %$ids;
+}
+
+=over 4
+
+=item create()
+
+Creates jobs given a list of settings passed as parameters. TEST setting/parameter
+is mandatory and should be the name of the test.
+
+=back
+
+=cut
+
+sub create {
+    my $self = shift;
+
+    my $global_params = $self->req->params->to_hash;
+    $self->{_is_clone_job} = delete $global_params->{is_clone_job} // 0;
+    my $grouped_params = _eval_param_grouping($global_params);
+    my $json = $self->{_json} = {};
+    my $event_data = $self->{_event_data} = [];
+    my $dependencies = $self->{_dependencies} = [];
     try {
-        my $downloads = create_downloads_list($job_settings);
-        my $schema = $self->schema;
-        $schema->txn_do(
+        $self->schema->txn_do(
             sub {
-                my $job = $schema->resultset('Jobs')->create_from_settings($job_settings);
-                my $job_id = $job->id;
-                $self->emit_event(openqa_job_create => {id => $job_id, %params});
-                $json->{id} = $job_id;
-
-                # enqueue gru jobs
-                push @{$downloads->{$_}}, [$job_id] for keys %$downloads;
-                $self->gru->enqueue_download_jobs($downloads);
-                $job->calculate_blocked_by;
-                OpenQA::Scheduler::Client->singleton->wakeup;
+                if (keys %$grouped_params) {
+                    # create as many jobs as unique ":"-suffixes exist
+                    $self->_create_job($global_params, $_, $grouped_params->{$_}) for keys %$grouped_params;
+                }
+                else {
+                    # create a single job if there are no job-specific parameters
+                    $self->_create_job($global_params);
```

Comment:
That additional level of indendation is a smell for me. How about making the additional parameters mandatory in `_create_job` and adding the single job parameter here as default in case `$grouped_params` is empty. Then you might be able to simplify the condition in https://github.com/os-autoinst/openQA/pull/4535/files#diff-6da53efb4105c5db577df2d5cd890fac0c8f83cd05a4b6a80bf5b067a4597fb2R307 as well

Reply:
Not sure whether I follow. Should the for-loop go into `_create_job` (which should then likely be named `_create_jobs_internally`)? I also don't see whether this helps simplifying the condition. The condition is about whether `ids` or `id` is populated and the latter is still required for keeping backwards compatibility. Only if there is at least one suffix (even if there's still just a single job¹) the `ids` key should be populated.

¹Otherwise it is more inconsistent in my opinion, e.g. I'd needed to distinguish between `id` and `ids` in the clone script.

Reply:
> Not sure whether I follow. Should the for-loop go into `_create_job` (which should then likely be named `_create_jobs_internally`)?

I think so, yes. I haven't tried this myself yet to be sure this would be better, sorry.

Reply:
Ok, I'll try it tomorrow and see whether it is better or not.

Reply:
Moved into a separate function.